System/SpatialPartitionGrid.cpp

```diff
@@ -146,29 +134,31 @@ namespace RTE {
 /////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
 	const std::vector<MovableObject *> & SpatialPartitionGrid::GetMOsInRadius(const Vector &centre, float radius, int ignoreTeam) const {
-		static std::vector<MovableObject *> s_MOList; // For script interop... :(
+		static std::vector<MovableObject *> s_MOList; // Note - This static vector exists to allow this data to be passed safely to Lua.
 		s_MOList.clear();
 
-		RTEAssert(ignoreTeam >= Activity::NoTeam && ignoreTeam < Activity::MaxTeamCount, "Invalid ignoreTeam given to GetMOsInRadius()!");
+		RTEAssert(ignoreTeam >= Activity::NoTeam && ignoreTeam < Activity::MaxTeamCount, "Invalid ignoreTeam given to SpatialPartitioningGrid::GetMOsInRadius()!");
 
 		std::unordered_set<MOID> potentialMOIDs;
 
-		int topLeftCellX = (centre.m_X - radius) / m_CellSize;
-		int topLeftCellY = (centre.m_Y - radius) / m_CellSize;
-		int bottomRightCellX = (centre.m_X + radius) / m_CellSize;
-		int bottomRightCellY = (centre.m_Y + radius) / m_CellSize;
+		int topLeftCellX = static_cast<int>(centre.m_X - radius) / m_CellSize;
```

Comment:
This is broken. Integer divide rounds towards 0, whereas we want to just throw away everything after the decimal. As such the cast to int needs to occur after the division, not before.